src/EventListener/ReadListener.php

```diff
@@ -33,24 +34,21 @@
  */
 final class ReadListener
 {
-    private $collectionDataProvider;
-    private $itemDataProvider;
-    private $subresourceDataProvider;
+    use OperationDataProviderTrait;
+
     private $serializerContextBuilder;
-    private $identifierDenormalizer;
 
     public function __construct(CollectionDataProviderInterface $collectionDataProvider, ItemDataProviderInterface $itemDataProvider, SubresourceDataProviderInterface $subresourceDataProvider = null, SerializerContextBuilderInterface $serializerContextBuilder = null, ChainIdentifierDenormalizer $identifierDenormalizer = null)
     {
         $this->collectionDataProvider = $collectionDataProvider;
         $this->itemDataProvider = $itemDataProvider;
         $this->subresourceDataProvider = $subresourceDataProvider;
         $this->serializerContextBuilder = $serializerContextBuilder;
+        $this->identifierDenormalizer = $identifierDenormalizer;
 
         if (null === $identifierDenormalizer) {
```

Comment:
Same here, I would not throw a deprecation if it's not provided. If you don't inject it, you'll just loose support for complex identifiers (not a big deal in many projects).